mllib/src/test/scala/org/apache/spark/ml/classification/ProbabilisticClassifierSuite.scala

```diff
@@ -91,4 +94,54 @@ object ProbabilisticClassifierSuite {
     "thresholds" -> Array(0.4, 0.6)
   )
 
+  def probabilisticClassifierGenericTest[
+      FeaturesType,
+      M <: ProbabilisticClassificationModel[FeaturesType, M]](
+    model: M, testData: Dataset[_]): Unit = {
+
+    val allColModel = model.copy(ParamMap.empty)
+      .setRawPredictionCol("rawPredictionAll")
+      .setProbabilityCol("probabilityAll")
+      .setPredictionCol("predictionAll")
+    val allColResult = allColModel.transform(testData)
+
+    for (rawPredictionCol <- Seq("", "rawPredictionSingle")) {
+      for (probabilityCol <- Seq("", "probabilitySingle")) {
```

Comment:
Just to confirm, does setting `probabilityCol`, `rawPredictionCol`, `predictionCol` to empty strings work here because expressions like `$(probabilityCol)` (used in [ProbabilisticClassifier.scala](https://github.com/apache/spark/blob/master/mllib/src/main/scala/org/apache/spark/ml/classification/ProbabilisticClassifier.scala#L115)) return the String value of probabilityCol?

Reply:
Yes.
And when set these 3 params to be empty string, they are regarded as turned off, according to the implementation in `ProbabilityClassificationModel`